### PR TITLE
Support hive querying as user for reads

### PR DESCRIPTION
**Pull request by billonahill**
This is written in a way that it can be submitted as a pull request to OSS presto with slight modifications around the commented part. It can also be enabled or disabled with a config flag.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java [hunk 1]
```diff
@@ -155,6 +157,28 @@ public void stop()
         @Override
         public TaskStatus process()
         {
+            if (HiveSessionProperties.getReadAsQueryUser(session)) {
+                try {
+                    // TODO: Configure hadoop to allow sqlsystem to impersonate all presto users (HADOOPINFRA-7081)
```
Comment (saileshmittal) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java]: s/sqlsystem/presto-daemon

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 public final class HiveSessionProperties
 {
     private static final String FORCE_LOCAL_SCHEDULING = "force_local_scheduling";
+    private static final String READ_AS_QUERY_USER = "read_as_query_user";
```
Comment (saileshmittal) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java]: Nit: These are probably in sorted order.

**Comment by saileshmittal** [comment]
@billonahill May be we should make it run as proxy user before merging?

**Comment by billonahill** [comment]
@saileshmittal if the proxy user settings get provisioned quickly then yes, I agree. If not I'd like to merge this after the new year just to verify performance and functionality without blocking on the proxy user request. Switching to run as proxy would then be a minor change once that's ready. Once we ship that change and run in prod for a bit we could then submit an upstream PR.

**Pull request merged**